### ISSUE TITLE
Should multiplicative_replacement raise a warning when Nans are generated?

### ISSUE THREAD

**Issue by johnchase** [biocore/scikit-bio#1241]
Running `multiplicative_replacement` in a table that contains a row of all zeros returns Nan for all of the items in that row.

```python 
import numpy as np
from skbio.stats.composition import multiplicative_replacement
X = np.array([[0,0,0,0],[0,.5,.5,0]])
multiplicative_replacement(X)

array([[    nan,     nan,     nan,     nan],
       [ 0.0625,  0.4375,  0.4375,  0.0625]])
```

This makes sense as there is no way to calculate a value for a row that only contains 0s, and returning zero defeats the purpose of running `multiplicative_replacement`.

I'm wondering if it may be valuable to raise a  warning though, as subsequently running `ancom` with this table will fail with a ValueError. And therefore requires the user to check the table being returned by `multiplicative_replacement`

If not perhaps the documentation should be updated? Currently the documentation reads:
"A matrix of proportions where all of the values are nonzero and each composition (row) adds up to 1" which is not strictly true.

@mortonjt you are far more familiar with the common usage and the implementation of this method, do you have thoughts on this?

**Comment by mortonjt** [comment]
That probably should raise an error - you shouldn't be able to pass in a vector with all zeros.

**Comment by johnchase** [comment]
This all looks good. Raising an error when a row of zeros is passed seems like a good solution for this issue. Thanks @mortonjt for looking into this!

**Comment by ebolyen** [comment]
Fixed in #1242
